numba/cuda/cudadrv/devicearray.py

```diff
@@ -115,6 +115,21 @@ def bind(self, stream=0):
         clone.stream = stream
         return clone
 
+    @property
+    def T(self):
+        return self.transpose(None)
+
+    def transpose(self, axes):
+        if self.ndim != 2:
+            raise NotImplementedError("transposing a non-2D DeviceNDArray isn't supported")
+        elif axes is not None and len(set(axes)) != len(axes):
+            raise ValueError("repeated axis in transpose")
+        elif axes == (0, 1):
+            return self
+        else:
```

Comment:
Invalid axes e.g. `(0, 2)` can still pass through. Can you add a test for such cases as well?